packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/AudienceView.tsx

```diff
@@ -25,21 +36,62 @@ export interface AudienceViewProps extends HasContainerId {
 export function AudienceView(props: AudienceViewProps): React.ReactElement {
 	const { containerId } = props;
 
-	// TODO: Post message requesting Audience summary
-	// TODO: Listen for Audience data updates
+	// POST Request for Audience Data
+	const getAudienceMessage: IDebuggerMessage = {
+		type: "GET_AUDIENCE",
+		source: extensionMessageSource,
+		data: {
+			containerId,
+		},
+	};
+
+	const messageRelay = useMessageRelay();
+
+	const [clientId, setClientId] = React.useState<string | undefined>("");
```

Comment:
Rather than storing these as 3 separate state variables, it might be easier to just store the data object we get from the message events. Since we always update at that granularity, there isn't much benefit to storing these independently.

If we set the default of this single state value to `undefined`, then we can display a spinner (see the `Waiting` component) until we get the initial data back from the debugger.

Reply:
This is awesome. Never thought about storing as an object, learned something new today! Fixed!